### PR TITLE
WIP: Test App templates

### LINKED ISSUE #28: Application Template support
Being able to run tests for a Blender Application Template would be a great feature for your module.

The main issues are:
- It will require some design changes.
- The name won't be accurate anymore.

### PR DESCRIPTION
**Pull request by Tilix4**
Hi!

I'm working on creating tests for Application templates. This is a first step and it works.   
The idea is: You can test an addon with an app template at the same time.

I renamed `addon` to `module` to use a name which includes both addon and application templates.

I also did some refactor, using less global variables and more functions attributes, which is more reliable and pythonic. (It's a freedom I took, it's up to you to tell me if it's consistent with your design).

I'm proposing you this code to make your work more global and valuable (it helped me a lot), and to keep everything in the same place instead of forking all the time.

Hope this first contribution of mine will foster you ;)

Closes #28

### CODE REVIEW

**Comment by douglaskastle** [comment]
Thanks for this.  I'll need to review.  

But first up is that none of the travis tests pass any more:

https://travis-ci.org/github/nangtani/blender-addon-tester/builds/746090718

Typically the branch needs to at least be passing in of itself to push a merge through.

**Comment by Tilix4** [comment]
Okay, I'll have a look to it!

**Comment by douglaskastle** [comment]
The testing is failing because these commands are no longer working.

    - cd ${TRAVIS_BUILD_DIR}
    - export PYTHONPATH=${TRAVIS_BUILD_DIR}
    - cd examples/testing-fake-addon/
    - python test_addon_blender.py ${BL_ADDON} ${BLENDER_VERSION}

Trying to debug further

**Comment by douglaskastle** [comment]
In the test script this is what is being called:

        exit_val = BAT.test_blender_addon(addon_path=addon, blender_revision=blender_rev)

which is calling this:
 
    from .run_blender import run_blender_version_for_addon_with_pytest_suite as test_blender_addon

which is this:
    
    def run_blender_version_for_addon_with_pytest_suite(addon_path="", app_template_path="", blender_exec_path="", blender_revision=None, config={}, dir_to_ignore=set()):

so the `app_template` is not being set explictly, so it is being set implicitly to '', which is this complaint:

    [Errno 2] No such file or directory: ''

@line 125 of `addon_helper.py`

I still don't understand all you have done to fix this yet.

**Comment by Tilix4** [comment]
Don't bother, I'll make it work, it's up to me :)

File: blender_addon_tester/blender_load_pytest.py [hunk 1]
```diff
@@ -35,29 +37,35 @@
...
 
     def pytest_unconfigure(self):
-        cleanup(self.addon, self.bpy_module, self.addon_dir)
+        cleanup(self.bpy_module, self.addon_dir, "APP_TEMPLATE")
         print("*** test run reporting finished")
 
 
 try:
-    pytest_main_args = [TESTS_PATH, "-v", "-x"]
+    pytest_main_args = ["/Users/felixdavid/Documents/Logiciels/Stax/stax/tests/", "-v", "-x"]
```
Comment (Tilix4) [inline blender_addon_tester/blender_load_pytest.py:63]: Really? I did it really dirty... Sorry for that.

**Comment by Tilix4** [comment]
Closed as it is too dirty. Will make a new one when it'll be ready.

**Pull request closed without merging**